components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -598,17 +601,23 @@ void AdsServiceImpl::OnInitialize(
     const int32_t result) {
   if (result != ads::Result::SUCCESS) {
     LOG(ERROR) << "Failed to initialize ads";
-    return;
+    is_initialized_ = false;
```

Comment:
Is there a code style requirement to fail early? As for me, the code is more clear when having less return statements.

Reply:
To be consistent with code in the ads_service and library as we follow the bouncer pattern

Reply:
@tmancey is the ads_service and library standalone elements or a part of brave-core? We should have a standard for a whole brave-core and not just because in the lib I did that way and I want it be that way. Otherwise if we don't have that on brave-core, it is up to an engineer which way to choose.

Reply:
https://bravesoftware.slack.com/archives/CC74MP8HW/p1569440886003800

Reply:
@SergeyZhukovsky the libraries are part of `brave-core` for Android/desktop and `standalone` for iOS

Reply:
If we are going that way, I would like to see a document that describes all that requirements for the code. Put it in the repo, put it in the wiki.

Reply:
NONCRITICAL
Covered by: https://github.com/brave/brave-browser/issues/6235 (1)